src/code-checks/code-checks.service.ts

```diff
@@ -0,0 +1,324 @@
+        const invalidFuelFlowPeriodCodes = getInvalidCodes<FuelFlowPeriodCode>(codesMap.get("fuelFlowPeriodCode"), FuelFlowPeriodCode.getRepository());
+        const invalidLtffUomCodes = getInvalidCodes<UnitsOfMeasureCode>(codesMap.get("longTermFuelFlowUnitsOfMeasureCode"), UnitsOfMeasureCode.getRepository());
+        const invalidGcvUomCodes = getInvalidCodes<UnitsOfMeasureCode>(codesMap.get("gcvUnitsOfMeasureCode"), UnitsOfMeasureCode.getRepository());
+        const invalidApsCodes = getInvalidCodes<ApsCode>(codesMap.get("apsCode"), ApsCode.getRepository());
+        const invalidSrcResultCodes = getInvalidCodes<TestResultCode>(codesMap.get("samplingRatioCheckResultCode"), TestResultCode.getRepository());
+        const invalidPlcResultCodes = getInvalidCodes<TestResultCode>(codesMap.get("postLeakCheckResultCode"), TestResultCode.getRepository());
+        const invalidTrainQaStatusCodes = getInvalidCodes<TrainQaStatusCode>(codesMap.get("trainQAStatusCode"), TrainQaStatusCode.getRepository());
+        const invalidCo2EmStdCodes = getInvalidCodes<Nsps4tEmissionStandardCode>(codesMap.get("co2EmissionStandardCode"), Nsps4tEmissionStandardCode.getRepository());
+        const invalidModusUomCodes = getInvalidCodes<UnitsOfMeasureCode>(codesMap.get("modusUnitsOfMeasureCode"), UnitsOfMeasureCode.getRepository());
+        const invalidElectricalLoadCodes = getInvalidCodes<Nsps4tElectricalLoadCode>(codesMap.get("electricalLoadCode"), Nsps4tElectricalLoadCode.getRepository());
+        const invalidAnnualEnergySoldTypeCodes = getInvalidCodes<Nsps4tElectricalLoadCode>(codesMap.get("annualEnergySoldTypeCode"), Nsps4tElectricalLoadCode.getRepository());
+        const invalidCo2RateUomCodes = getInvalidCodes<TrainQaStatusCode>(codesMap.get("co2EmissionRateUnitsOfMeasureCode"), TrainQaStatusCode.getRepository());
+
+        try {
+            const results = await Promise.all([invalidParameterCode, invalidFuelCodes, invalidTestTypeCodes,
+                invalidTestResultCodes, invalidSpanScaleCodes, invalidGasLevelCodes, invalidUpscaleGasLevelCodes,
+                invalidInjProtocolCodes, invalidVendors, invalidLoadUomCodes, invalidModcCodes, invalidOpCondCodes,
+                invalidVolUomCodes, invalidSodVolCodes, invalidSodMassCodes, invalidSampleCodes, invalidParamUomCodes,
+                invalidHourFlags, invalidFuelFlowPeriodCodes, invalidLtffUomCodes, invalidGcvUomCodes, invalidApsCodes,
+                invalidSrcResultCodes, invalidPlcResultCodes, invalidTrainQaStatusCodes, invalidCo2EmStdCodes,
+                invalidModusUomCodes, invalidElectricalLoadCodes, invalidAnnualEnergySoldTypeCodes, invalidCo2RateUomCodes])
+
+            // The results of the query come back with a list of invalid codes which are used to generate the error msgs                                    
+            const parameterCodeErrors = results[0].map(invalidCodeObj => getErrorMessage("parameterCode", invalidCodeObj.column1))
```

Comment:
Then this can be...
const parameterCodeErrors = results[PARAMETER_CODES].map(invalidCodeObj => getErrorMessage("parameterCode", invalidCodeObj.column1))